free-1.md

```diff
@@ -57,12 +95,25 @@ making language design decisions), as opposed to "systems" aspects of the
 project (e.g., implementing a complicated semantics that doesn't require a lot
 of language design)?_
 
+Because programs in this language have such simple syntax, I imagine most of someone's time spent on thier project would be working on the "systems" aspect of the project. However, because the lanugage (at least right now) has natural language (English) embedded in it, building the parser (which I consider part of the language aspects of the project) could be a difficult thing to do.
+
 
 ### Scope
 _How big an idea is this? How ambitious is this project?_
 
+While the language itself will be pretty small, the project would be fairly difficult to actually implement. This is largely because creating and implementing the models used on the backend (the agggregator for student choices, and the pre-registrator code) would be fairly difficult. However, I think that an impementer could successfully implement this project end-to-end in half a semester. 
 
 ### Benefits and drawbacks
 _Why might this be a good idea for a project? Why might this not be a good idea 
 project?_
 
+I think this would be a good idea for a project because it directly solves a problem experienced by the immediate population. 
+The benefits of this are twofold: 
+1. The implementer can easily gather user requirements/feature requests as they build the project.
+2. We know for sure that there will be a user base for this DSL. 
+
+There are also two main reasons why this might not be a good project idea:
+1. As mentioned earlier, the implementer would probably spend a lot more time engaging in the systems aspect of the language than the design aspect. Since this class seems to be focused on language design in a big way, this project may not be best at fulfilling the learning goals of the class.
+2. Even if this language did exist, the bureaucracy required to get anything done with Mudd's administration may be a prohobitively large barrier to actually adopting this system.
+
+I think the main tradeoff in terms of deciding whether to implement this project is the tradeoff between spending a lot of time on language design (which is more in line with this class's goals) and building something that would be really useful to Mudders. 
```

Comment:
I think this idea would make for a really interesting project. As you mentioned, the language-design aspect of may be a little shallow, but there are usually some unexpected language-design surprises that come up.

Also, this is a specific instance of a constraint-satisfaction problem. Constraint-satisfaction problems are everywhere, and a general DSL for smaller instances of such problems would be nice. So, one potential path for this project is as follows:

   1. Develop the syntax for the language you proposed, to solve the course-registration problem.
   1. For the semantics, see if you can compile / translate the program to an existing constraint-satisfaction library or tool.
   1. At this point, you're done! That might take the rest of the semester. But, if not, you can then work on generalizing your language: can it solve other constraint problems?